File: javascript/3355.js
/**
 * @param {number[]} n
 * @param {number[][]} q
 * @return {boolean}
 */
var isZeroArray = function (n, q) {
    let l = n.length, p = 0;
    let r = new Array(l + 1).fill(0);
    for (let i of q) {
        r[i[0]]++;
        r[i[1] + 1]--;
    }
    for (let i = 0; i < l; i++) {
        p += r[i];
        if (n[i] > p)
            return false;
    }
    return true;
};

File: cpp/3355.cpp
class Solution {
public:
    bool isZeroArray(vector<int>& n, vector<vector<int>>& q) {
        int l = n.size();
        vector<int> r(l + 1);
        for (vector<int>& i : q) {
            r[i[0]]++;
            r[i[1] + 1]--;
        }
        int p = 0;
        for (int i = 0; i < l; i++) {
            p += r[i];
            if (n[i] > p)
                return false;
        }
        return true;
    }
};
